Dockerfile

```diff
@@ -4,13 +4,15 @@ LABEL org.opencontainers.image.source https://github.com/opensafely/research-tem
 
 # we are going to use an apt cache on the host, so disable the default debian
 # docker clean up that deletes that cache on every apt install
-RUN rm -f /etc/apt/apt.conf.d/docker-clean
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
 
-# DL3042: we always want latest package versions when we rebuild
+# DL3008, DL3042: we always want latest package versions when we rebuild
+# DL3009: we are caching apt
 # DL3013: using an apt cache on the host instead
-# hadolint ignore=DL3042,DL3013
+# hadolint ignore=DL3008,DL3009,DL3042,DL3013
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-RUN --mount=type=cache,target=/var/cache/apt \
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
```

Comment:
I'm still seeing a DL3008 warning on GitHub on the [Files changed](https://github.com/opensafely-core/research-template-docker/pull/42/files) view for this PR

Reply:
Ah, maybe the comment is in the wrong place?

Reply:
Looks like whatever you've just pushed has worked!

Reply:
Yep, think this is done now, unless there's anything else.